pkg/kademlia/kademlia.go

```diff
@@ -329,6 +329,9 @@ func Restrict(r pb.Restriction, n []*pb.Node) []*pb.Node {
 
 func meetsRestrictions(rs []pb.Restriction, n pb.Node) bool {
 	for _, r := range rs {
+		if proto.Equal(&r, &pb.Restriction{}) {
+			return true
```

Comment:
Why should this return true?

Reply:
If an empty restriction was passed through, it caused no nodes to match. We want the opposite behavior.

Reply:
I mean, why not pass no restrictions for that case rather than passing in the empty restriction?

Reply:
I think it's better to be explicit than implicit in this case. It's not like it's less performant or less clear.

Reply:
To me it's inconsistent. The previous constraint checking was to `&` together all the restrictions together, this is clear. However now it's `&` except when you have case Z.

Also compare:
```
srv.dht.GetNodes(ctx, srv.identity.ID, 5, pb.Restriction{...}) // with a restriction
srv.dht.GetNodes(ctx, srv.identity.ID, 5) // without restrictions
```

The latter is shorter and quite clear that there isn't a restriction in place.